libosmscout/src/osmscout/routing/RouteNodeDataFile.cpp

```diff
@@ -0,0 +1,181 @@
+/*
+  This source is part of the libosmscout library
+  Copyright (C) 2018  Tim Teulings
+
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation; either
+  version 2.1 of the License, or (at your option) any later version.
+
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307  USA
+*/
+
+#include <osmscout/routing/RouteNodeDataFile.h>
+
+namespace osmscout {
+
+  RouteNodeDataFile::RouteNodeDataFile(const std::string& datafile,
+                           size_t cacheSize)
+  : datafile(datafile),
+    cache(cacheSize),
+    tileCalculator(0)
+  {
+  }
+
+  bool RouteNodeDataFile::Open(const TypeConfigRef& typeConfig,
+                         const std::string& path,
+                         bool memoryMappedData)
+  {
+    this->typeConfig=typeConfig;
+
+    datafilename=AppendFileToDir(path,datafile);
+
+    try {
+      FileOffset indexFileOffset;
+      uint32_t   dataCount;
+      uint32_t   indexEntryCount;
+      uint32_t   tileMag;
+
+      scanner.Open(datafilename,
+                   FileScanner::LowMemRandom,
+                   memoryMappedData);
+
+      scanner.Read(indexFileOffset);
```

Comment:
current router.dat (database format 16) don't contains any index right? This is causing crash with segfault with already existing databases!

Reply:
Right, I have to increase the data version.